darshan-runtime/lib/darshan-mpiio.c

```diff
@@ -1899,6 +1899,688 @@ static void mpiio_cleanup()
+    tm1 = MPIIO_WTIME();
+    ret = __real_PMPI_File_iread_all_c(fh, buf, count, datatype, request);
+    tm2 = MPIIO_WTIME();
+
+    MPIIO_PRE_RECORD();
+    MPIIO_RECORD_READ(ret, fh, count, datatype, offset, MPIIO_NB_READS, tm1, tm2);
+    MPIIO_POST_RECORD();
+
+    return(ret);
+}
+DARSHAN_WRAPPER_MAP(PMPI_File_iread_all_c, int, (MPI_File fh, void * buf, MPI_Count count, MPI_Datatype datatype, __D_MPI_REQUEST * request),
+        MPI_File_iread_all_c)
+
+int DARSHAN_DECL(MPI_File_iread_at_all_c)(MPI_File fh, MPI_Offset offset, void * buf,
+    MPI_Count count, MPI_Datatype datatype, __D_MPI_REQUEST *request)
+{
+    int ret;
+    double tm1, tm2;
+
+    MAP_OR_FAIL(PMPI_File_iread_at_all_c);
+
+    tm1 = MPIIO_WTIME();
+    ret = __real_PMPI_File_iread_at_all_c(fh, offset, buf, count,
+        datatype, request);
+    tm2 = MPIIO_WTIME();
+
+    MPIIO_PRE_RECORD();
+    MPIIO_RECORD_READ(ret, fh, count, datatype, offset, MPIIO_NB_READS, tm1, tm2);
+    MPIIO_POST_RECORD();
+
+    return(ret);
+}
+DARSHAN_WRAPPER_MAP(PMPI_File_iread_at_all_c, int, (MPI_File fh, MPI_Offset offset, void * buf,
+    MPI_Count count, MPI_Datatype datatype, __D_MPI_REQUEST *request),
+        MPI_File_iread_at_all_c)
+
+int DARSHAN_DECL(MPI_File_iread_at_c)(MPI_File fh, MPI_Offset offset, void * buf,
+    MPI_Count count, MPI_Datatype datatype, __D_MPI_REQUEST *request)
+{
+    int ret;
+    double tm1, tm2;
+
+    MAP_OR_FAIL(PMPI_File_iread_at_c);
+
+    tm1 = MPIIO_WTIME();
+    ret = __real_PMPI_File_iread_at_c(fh, offset, buf, count,
+        datatype, request);
+    tm2 = MPIIO_WTIME();
+
+    MPIIO_PRE_RECORD();
+    MPIIO_RECORD_READ(ret, fh, count, datatype, offset, MPIIO_NB_READS, tm1, tm2);
+    MPIIO_POST_RECORD();
+
+    return(ret);
+}
+DARSHAN_WRAPPER_MAP(PMPI_File_iread_at_c, int, (MPI_File fh, MPI_Offset offset, void * buf,
+    MPI_Count count, MPI_Datatype datatype, __D_MPI_REQUEST *request),
+        MPI_File_iread_at_c)
+
+int DARSHAN_DECL(MPI_File_iread_c)(MPI_File fh, void * buf, MPI_Count count, MPI_Datatype datatype, __D_MPI_REQUEST * request)
+{
+    int ret;
+    double tm1, tm2;
+    MPI_Offset offset;
+
+    MAP_OR_FAIL(PMPI_File_iread_c);
+
+    MPI_File_get_position(fh, &offset);
+    tm1 = MPIIO_WTIME();
+    ret = __real_PMPI_File_iread_c(fh, buf, count, datatype, request);
+    tm2 = MPIIO_WTIME();
+
+    MPIIO_PRE_RECORD();
+    MPIIO_RECORD_READ(ret, fh, count, datatype, offset, MPIIO_NB_READS, tm1, tm2);
+    MPIIO_POST_RECORD();
+
+    return(ret);
+}
+DARSHAN_WRAPPER_MAP(PMPI_File_iread_c, int, (MPI_File fh, void * buf, MPI_Count count, MPI_Datatype datatype, __D_MPI_REQUEST * request),
+        MPI_File_iread_c)
+
+int DARSHAN_DECL(MPI_File_iread_shared_c)(MPI_File fh, void * buf, MPI_Count count,
+    MPI_Datatype datatype, __D_MPI_REQUEST * request)
+{
+    int ret;
+    double tm1, tm2;
+    MPI_Offset offset;
+
+    MAP_OR_FAIL(PMPI_File_iread_shared_c);
+
+    MPI_File_get_position(fh, &offset);
```

Comment:
This one should be a `MPI_File_get_position_shared()`, right?

Reply:
Thanks for catching this error.